src/shogun/classifier/mkl/MKL.cpp

```diff
@@ -271,7 +271,7 @@ void CMKL::register_params()
 	rho = 0;
 	lp_initialized = false;
 
-	SG_ADD((CSGObject**) &svm, "svm", "wrapper svm", MS_NOT_AVAILABLE);
+	SG_ADD((CMachine**)&svm, "svm", "wrapper svm", MS_NOT_AVAILABLE);
```

Comment:
This is done as we want to register members by base type.
Here is an issue with this: users can now set **any** machine type to mkl.
This will lead to a runtime error, when the machine is used as an svm